src/Altinn.App.Core/Infrastructure/Clients/Storage/DataClient.cs

```diff
@@ -87,12 +94,15 @@ public async Task<DataElement> InsertFormData<T>(
     )
         where T : notnull
     {
+        using var timeoutCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+        timeoutCts.CancelAfter(_httpOperationTimeout);
+
```

Comment:
This approach of combining tokens in all the methods is kind of noisy. Could we rather inject `IHttpClientFactory` and use GetClient in all the methods, with a custom timeout argument to be used when we need it overridden?